[FEATURE] Bloquer l'accès à Pix Certif sur une période donnée (PIX-19092).

yaf opened the pull request:
## 🔆 Problème

Nous allons avoir une période pendant laquelle il ne faudra pas lancer de session de certification pour pouvoir faire la mise à jour des calibrations (et l'historisation des anciennes).

## ⛱️ Proposition

Utiliser l'écran de blocage existant avec une nouvelle variable d'environnement pour déclencher le blocage des manipulations de session dans pixCertif.

<img width="1725" height="902" alt="Capture d’écran 2025-09-09 à 15 56 34" src="https://github.com/user-attachments/assets/0fc6f303-40de-4bc1-9545-9c2f25d5a5bd" />


## 🌊 Remarques

L'accès à l'espace surveillant concerne 2 types d'utilisateurs différents : 
- l'utilisateur membre d'un centre de certification
- l'utilisateur non membre d'un centre de certification.

C'est important de le garder en tête car le blocage de Pix Certif coté centre de certif se fait via la route `/me` qui va répertorier les centres auxquels l'utilisateur est membre et retourner si le centre en question est bloqué ou non.

Mais un utilisateur non membre d'un centre aura un` /me` sans aucun centre.

On ne peut donc pas profiter des infos du `/me` pour bloquer l'utilisateur coté espace surveillant.

J'ai donc fait le choix (dernier commit) de jeter l'utilisateur coté `/supervise`

## 🏄 Pour tester

### Non reg

* Tester de se tromper sur le code d'accès candidat à la session
* Tenter de rentrer dans une session finalisée en tant que candidat
* Tenter de rentrer en tant que surveillant dans une session finalisée

### Nouvelle feature

La date de fin de période de la variable d'environnement `PIX_CERTIF_BLOCKED_ACCESS_UNTIL_DATE=2025-10-16`;
La whitelist permet l'accès au centre dont l'id est 1 (PRO Certification Center) `PIX_CERTIF_BLOCKED_ACCESS_WHITELIST=1`

- Se rendre sur [PixCertif](https://certif-pr13469.review.pix.fr/) avec certif-prescriptor@example.net et constater que le centre PRO Certification Center est disponible
- En changeant de centre en bas à gauche, constater que c'est bloqué pour tous les autres centres.
- essayer plusieurs url pour forcer l'accès : à la création d'une session, création de session en masse https://certif-pr13469.review.pix.fr/sessions/import)

- Aller sur l'espace surveillant et rejoindre la session 7402 / 3CwbfW (il s'agit d'une session du centre dans la whitelist)
- => on peut y accéder
- Quitter et tenter de rejoindre la session 7400 / QJy7mX => se faire jeter avec le message de maintenance.

<img width="414" height="757" alt="Capture d’écran 2025-09-09 à 14 22 58" src="https://github.com/user-attachments/assets/2dfb4ad1-1503-4c11-9ba7-4cf53152b866" />

- Modifier de nouveau la date de fin de période de la variable d'environnement `PIX_CERTIF_BLOCKED_ACCESS_UNTIL_DATE=2020-01-01`;

=> On doit pouvoir avoir accès à tout.

pix-bot-github commented:
Choisir les applications à déployer :

- [x] [API](https://api-pr13469.review.pix.fr/api/) | [👩‍💻 Dashboard Scalingo](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-review-pr13469/environment) <!-- pix-api-review -->
- [x] [App (.fr)](https://app-pr13469.review.pix.fr) – [App (.org)](https://app-pr13469.review.pix.org) | [👩‍💻 Dashboard Scalingo](https://dashboard.scalingo.com/apps/osc-fr1/pix-app-review-pr13469/environment) <!-- pix-app-review -->
- [ ] [Orga (.fr)](https://orga-pr13469.review.pix.fr) – [Orga (.org)](https://orga-pr13469.review.pix.org) | [👩‍💻 Dashboard Scalingo](https://dashboard.scalingo.com/apps/osc-fr1/pix-orga-review-pr13469/environment) <!-- pix-orga-review -->
- [x] [Certif (.fr)](https://certif-pr13469.review.pix.fr) – [Certif (.org)](https://certif-pr13469.review.pix.org) | [👩‍💻 Dashboard Scalingo](https://dashboard.scalingo.com/apps/osc-fr1/pix-certif-review-pr13469/environment) <!-- pix-certif-review -->
- [ ] [Junior](https://junior-pr13469.review.pix.fr) | [👩‍💻 Dashboard Scalingo](https://dashboard.scalingo.com/apps/osc-fr1/pix-junior-review-pr13469/environment) <!-- pix-junior-review -->
- [x] [Admin](https://admin-pr13469.review.pix.fr) | [👩‍💻 Dashboard Scalingo](https://dashboard.scalingo.com/apps/osc-fr1/pix-admin-review-pr13469/environment) <!-- pix-admin-review -->
- [ ] [API MaDDo](https://pix-api-maddo-review-pr13469.osc-fr1.scalingo.io/api/) | [👩‍💻 Dashboard Scalingo](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-maddo-review-pr13469/environment) <!-- pix-api-maddo-review -->
- [ ] [Audit Logger](https://pix-audit-logger-review-pr13469.osc-fr1.scalingo.io/api/) | [👩‍💻 Dashboard Scalingo](https://dashboard.scalingo.com/apps/osc-fr1/pix-audit-logger-review-pr13469/environment) <!-- pix-audit-logger-review -->

> [!IMPORTANT]
> N'oubliez pas de déployer l'API pour pouvoir accéder aux fronts et/ou à l’API MaDDo.

Steph0 commented on api/src/identity-access-management/domain/read-models/AllowedCertificationCenterAccess.js:
```diff
@@ -38,6 +38,10 @@ class AllowedCertificationCenterAccess {
   }
 
+  isAccessBlockedForPeriod() {
+    return !this.isInWhitelist() && new Date() < new Date(features.pixCertifScoBlockedAccessDateForPeriod);
```
Nous avions discute des elements suivants a l'oral que je ne vois pas ici

* Inversion du `new Date() < new Date` , on veut bloquer "a partir de"
* Ajouter une date de fin (soit deux var d'env, soi un split fait dans config.js via un separator)

Je ne comprends pas aussi pourquoi il y a la whitelist dans ce code. La whitelist est une specificite uniquement au SCO. La feature que tu implementes a pour but de bloquer tout le monde sans exceptions.

Steph0 commented on api/src/identity-access-management/domain/read-models/AllowedCertificationCenterAccess.js:
```diff
@@ -73,6 +77,10 @@ class AllowedCertificationCenterAccess {
     return features.pixCertifScoBlockedAccessDateCollege ?? null;
   }
+
+  get pixCertifScoBlockedAccessDateForPeriod() {
```
Je comprends pas a quoi sert ceci par rapport a l'autre fonction 🤔

yaf replied:
Ah si, c'est pour « enmener » la date de blocage jusqu'au front. Mais avec l'option que l'on a prise, nous ne savons pas quand ça va être remis en route :-/ Donc ça ne sert à rien. Mais il faudra peut-être discuter avec le métier, parce que ça change de ce qui est demandé sur le ticket...

C'est peut-être mieux de ne pas préciser quand ça va re-ouvrir. Mais en même temps, niveau UX, c'est pas hyper sympa pour l'utilisateur

yaf replied:
Vu avec @Agnes-V , c'est mieux que ce soit la date de fin, celle qui annonce la réouverture.

https://1024pix.slack.com/archives/CHGPQ7W9W/p1757085418245649

Steph0 commented:
⚠️ Le ticket mentionne d'ajouter le blocage de l'espace surveillant. L'acces a Pix Certif est bloque mais je peux toujours entrer en session, il y a donc une faille dans les blocages d'acces.

<img width="981" height="988" alt="Screenshot_20250905_165444" src="https://github.com/user-attachments/assets/b8464a9d-4564-4606-9ed5-63dcc3f69742" />

yaf commented:
Le ticket mentionne le blocage de l'espace surveillant, mais il y a un autre ticket qui l'évoque aussi. J'ai préféré séparer. Je vais modifier ce ticket là.

cf https://1024pix.atlassian.net/jira/software/c/projects/PIX/boards/91?selectedIssue=PIX-19390

En relisant, je constate qu'en fait on parle de « l'accès à l'espace surveillant » au sens du menu je suppose. Je laisse le ticket tel quel. Mais le blocage dans l'espace surveillant sera à faire dans une pr à suivre.

yaf commented:
En fait, c'est bien dans cette PR qu'il faut ajouter le blocage à l'espace surveillant pour tout le monde.

L'autre ticket concerne un manque lié au blocage dans le cadre du SCO... 

Il reste donc un blocage à mettre en place ici.

yaf commented on certif/app/routes/session-supervising.js:
```diff
@@ -5,11 +5,16 @@ import ENV from 'pix-certif/config/environment';
 
+  beforeModel() {
+    this.currentUser.checkRestrictedAccess();
+  }
```
j'ai l'impression que cet ajout, sous cette forme, permet de traiter aussi le cas du ticket https://1024pix.atlassian.net/browse/PIX-19390 ...

AndreiaPena replied:
je me suis permise de l'enlever car :

- pour l'utilisateur membre d'un centre de certif, plutôt que de jeter une erreur coté formulaire espace surveillant, ça redirigeait vers l'espace pix certif (pas souhaité)
- pour l'utilisateur non membre d'un centre de certif, ça ne peut pas fonctionner car `checkRestrictedAccess` se fie aux centres auxquels un utilisateur est membre.

yaf replied:
Mais du coup, ça ne bloque plus l'espace surveillant ?

AndreiaPena reviewed (commented):
<img width="1723" height="590" alt="Capture d’écran 2025-09-08 à 16 16 44" src="https://github.com/user-attachments/assets/bf53d86b-0f7d-46de-9c34-244132815de6" />

J'obtiens bien la page souhaitée, en revanche le wording est différent selon le ticket Jira. On s'attend à avoir **"Site en maintenance, réouverture de votre espace Pix Certif le 01/..."** 🔴

AndreiaPena commented:
https://certif-pr13469.review.pix.fr/sessions/import

On peut toujours accéder à l'import en masse de session, même si l'espace est fermé :/ 
Je me demande si c'est pas aussi le cas lorsque les centres SCO sont fermés.

yaf commented:
> https://certif-pr13469.review.pix.fr/sessions/import 🔴
> 
> On peut toujours accéder à l'import en masse de session, même si l'espace est fermé :/ Je me demande si c'est pas aussi le cas lorsque les centres SCO sont fermés.

Comme nous avons re-utilisé la mécanique existante, il est fort probable que le problème existe déjà pour le SCO. Mais on va faire en sorte que ça bloque pour les deux :)

Agnes-V commented:
Hello @yaf !
Pour faire suite au com d'@AndreiaPena : 
Si c'est possible sans trop de modifs de rajouter le wording initialement prévu ça serait top.
En effet, même si on réutilise les écrans SCO rationaliser un max, on n'est pas exactement dans la même situation côté utilisateur.
Dans le cas du SCO, ils connaissent la mécanique et le calendrier scolaire, ça n'est donc pas une surprise pour eux que leur espace Pix Certif soit bloqué.
En revanche, dans le cas de la période blanche, on est dans le cadre de l'exceptionnel. Si l'utilisateur ne comprend pas pourquoi il voit ce message (risque si on ne mentionne pas qu'il y a une maintenance en cours), j'ai peur qu'ils contactent en masse le Support pour des explications ou des demandes de déblocage. 😓
D'ailleurs ça me fait penser que contrairement au SCO, on risque d'avoir des utilisateurs autres que francophones. Il faut donc que ce message affiché pendant la période blanche soit : 
- FR : "**Site en maintenance, réouverture de votre espace Pix Certif le 04/10/2025**"
- EN : “**Site under maintenance, your Pix Certif space will reopen on 04/10/2025.**” 🙏

Agnes-V commented:
@yaf et @AndreiaPena pour ce qui est du commentaire sur la gestion massive des sessions, cette fonctionnalité n'est pas activée pour les centres SCO, d'où le trou dans la raquette j'imagine 🏸

yaf commented:
@Agnes-V  pour la traduction, il faut localiser la date en anglais aussi j'imagine ? 10/04/2025 au lieu de 04/10/2025 de l'exemple

yaf commented:
> 
> J'ai pu me connecter à l'espace surveillant => 7402 / `3CwbfW` 🔴

Je n'ai pas réussi à reproduire le truc. C'est peut-être parce que l'application Certif n'avait pas redémarré ? ou l'API ?

yaf commented:
> https://certif-pr13469.review.pix.fr/sessions/import 🔴
> 
> On peut toujours accéder à l'import en masse de session, même si l'espace est fermé :/ Je me demande si c'est pas aussi le cas lorsque les centres SCO sont fermés.

:heavy_check_mark: C'est corrigé normalement.

Agnes-V commented:
@yaf pour la date on reste sur le même affichage 04/10/2025 car c'est de l'anglais britannique sur Pix, seuls les américains inversent 🙃

AndreiaPena committed:
✨ api: block pix certif access

Co-authored-by: Yannick François <yannick.francois@pix.fr>

AndreiaPena committed:
✨ certif: block pix certif access

Co-authored-by: Yannick François <yannick.francois@pix.fr>

AndreiaPena committed:
✨ api: block invigilator access

AndreiaPena committed:
✨ certif: block invigilator access

AndreiaPena commented on api/src/certification/session-management/domain/usecases/supervise-session.js:
```diff
@@ -1,27 +1,44 @@
+      }).replaceAll('&#x2F;', '/'),
+      'PIX_CERTIF_ACCESS_BLOCKED',
+    );
+  }
```
(voir la partie "Remarques" dans la PR avant)

Le modèle session porte les infos de la date de blocage. Plutôt que de transmettre la date au front, j'ai trouvé plus simple de gérer le message d'erreur via l'api direct, que le front va simplement rapporter à l'utilisateur.

J'ai pas cherché plus mais le `replaceAll` c'est simplement parce que le translate casse le formattage de la date :'(

Agnes-V commented:
Hello !
Review func : 
- Pix Certif .fr ✅
<img width="1423" height="700" alt="Capture d’écran 2025-09-09 à 17 16 56" src="https://github.com/user-attachments/assets/e28cb38c-a2b1-40fa-9cb7-a343350e78fe" />
- Espace surveillant .fr ✅
<img width="1423" height="764" alt="Capture d’écran 2025-09-09 à 17 16 45" src="https://github.com/user-attachments/assets/6de52f80-49ac-4739-b155-4420ceebca8e" />

En revanche :
- ⚠️ date de réouverture indiquée dans le message 10/10/2025, ce qui n'est pas bon. En réalité il faudra indiquer le 04/10/2025 mais c'est peut-être normal et une date au pif qui a été choisie, qui sera précisée le moment venu. 🤷🏽‍♀️ Si c'est le cas, vous pouvez ignorer ce retour 🙈
- ❌ Pix Certif .org je n'ai pas pu y accéder avec les seeds habituels `certif-prescriptor@example.net` (MDP vérifié aussi), est-ce qu'il y a une manip en particulier à faire pour accéder à Pix Certif en .org ?
<img width="1423" height="700" alt="Capture d’écran 2025-09-09 à 17 22 21" src="https://github.com/user-attachments/assets/6e2199da-3792-4343-9c3f-2e29bafaf287" />

AndreiaPena commented:
> date de réouverture indiquée dans le message 10/10/2025, ce qui n'est pas bon. En réalité il faudra indiquer le 04/10/2025 mais c'est peut-être normal et une date au pif qui a été choisie, qui sera précisée le moment venu. 🤷🏽‍♀️ Si c'est le cas, vous pouvez ignorer ce retour 🙈

@Agnes-V en effet la date est modifiable, c'est une variable d'environnement :)

Agnes-V commented:
Test func OK sur Certif.org (message d'erreur en français mais présent) ✅
<img width="1009" height="750" alt="Capture d’écran 2025-09-11 à 10 32 04" src="https://github.com/user-attachments/assets/7cf8c33c-7d68-4051-accf-0b4544bf7041" />

AndreiaPena commented on api/src/identity-access-management/domain/read-models/AllowedCertificationCenterAccess.js:
```diff
@@ -77,10 +77,6 @@ class AllowedCertificationCenterAccess {
     return features.pixCertifScoBlockedAccessDateCollege ?? null;
   }
-
-  get pixCertifBlockedAccessUntilDate() {
```
Ce getter est utilisé par le serializer, il est bien utilisé

lego-technix commented on api/src/certification/session-management/domain/usecases/supervise-session.js:
```diff
@@ -2,26 +2,41 @@ import {
   CertificationCenterIsArchivedError,
   InvalidSessionSupervisingLoginError,
   SessionNotAccessible,
-} from '../../domain/errors.js';
+} from '../errors.js';
 
 const superviseSession = async function ({
   sessionId,
   invigilatorPassword,
   userId,
+  i18n,
   sessionRepository,
   supervisorAccessRepository,
   certificationCenterRepository,
 }) {
   // should use a specific get from sessionRepository instead
   const session = await sessionRepository.get({ id: sessionId });
 
+  if (session.isAccessBlocked()) {
+    const formattedDate = new Intl.DateTimeFormat('fr-FR').format(new Date(session.pixCertifBlockedAccessDate));
+    const translate = i18n.__;
+    throw new SessionNotAccessible(
+      translate('certification.error-messages.pix-certif-access', {
+        date: formattedDate,
+      }).replaceAll('&#x2F;', '/'),
+      'PIX_CERTIF_ACCESS_BLOCKED',
+    );
+  }
```
Comme présenté lors de la MeJ sur la _gestion des locales_, l’équipe Accès recommande désormais de ne plus passer `i18n` aux usecases, mais de passer la locale aux usecases et de faire l’appel `const i18n = getI18n(locale);` dans les usecases. Ce qui donnerait : 

```suggestion
const superviseSession = async function ({
  sessionId,
  invigilatorPassword,
  userId,
  locale,
  sessionRepository,
  supervisorAccessRepository,
  certificationCenterRepository,
}) {
  // should use a specific get from sessionRepository instead
  const session = await sessionRepository.get({ id: sessionId });

  if (session.isAccessBlocked()) {
    const formattedDate = new Intl.DateTimeFormat(locale).format(new Date(session.pixCertifBlockedAccessDate));
    const translate = i18n.__;
    throw new SessionNotAccessible(
      translate('certification.error-messages.pix-certif-access', {
        date: formattedDate,
      }).replaceAll('&#x2F;', '/'),
      'PIX_CERTIF_ACCESS_BLOCKED',
    );
  }
```

Nous n’avons pas encore appliqué cette recommandation dans le code mais ce sera très bientôt, alors autant que ça soit déjà le cas ici 😃

lego-technix commented on api/src/certification/session-management/application/supervise-controller.js:
```diff
@@ -1,9 +1,13 @@
+  const locale = request.query?.lang;
+  const i18n = getI18n(locale);
+
+  await usecases.superviseSession({ sessionId, userId, invigilatorPassword, i18n });
```
L’utilitaire `getUserLocale` sait récupérer au mieux la locale de l’utilisateur dans toutes les configurations possibles :

```suggestion
  import { getUserLocale } from '../../../shared/infrastructure/utils/request-response-utils.js';
  const locale = getUserLocale(request);

  await usecases.superviseSession({ sessionId, userId, invigilatorPassword, locale });
```

AndreiaPena replied:
Merci pour l'info ! Avec ton retour sur le fait d'envoyer la date en meta de l'erreur, je vais enlever l'usage du intl ^^

lego-technix commented on api/src/certification/session-management/domain/usecases/supervise-session.js:
```diff
@@ -2,26 +2,41 @@ import {
+      'PIX_CERTIF_ACCESS_BLOCKED',
+    );
+  }
+
```
Ceci dit, juste une remarque non spécifique au périmètre de l’équipe Accès : pourquoi gérer la gestion de l’affichage du message dans l’API et pas dans le frontal ? cf. https://github.com/1024pix/pix/blob/dev/docs/adr/0044-gestion-erreurs-i18n-reference.md Cela serait cohérent avec la gestion des autres cas par des exceptions. Et cela fait une API « plus API », « plus machine », et des frontaux qui s’occupent de l’affichage pour les utilisateurs humains.

Suggestion : 
```javascript
if (session.isAccessBlocked()) {
  throw new SessionNotAccessible(session.pixCertifBlockedAccessDate);
}
```

Si cette suggestion vous semble appropriée, alors le constructeur de `SessionNotAccessible` peut être modifié ainsi, en effet il n’y a aucune raison de donner la possibilité de modifier le `message` ni le `code` dans le constructeur :
```javascript
class SessionNotAccessible extends DomainError {
  constructor(blockedAccessDate) {
    super('Certification session is not accessible', 'SESSION_NOT_ACCESSIBLE');
    if (blockedAccessDate) {
        this.meta = { blockedAccessDate };
    }
  }
}
```

AndreiaPena replied:
Cette suggestion me convient très bien, merci ! :D je prend !

Steph0 committed:
✨ api: add whitelist for access blocking feature

Add PIX_CERTIF_BLOCKED_ACCESS_WHITELIST environment variable to allow
specific certification centers to bypass the general blocking period
defined by PIX_CERTIF_BLOCKED_ACCESS_UNTIL_DATE.

AndreiaPena commented on api/sample.env:
```diff
@@ -1130,3 +1130,13 @@ DD/MM/YYYY [-] HH:mm:ss [[].SSS]
+# Linked to PIX_CERTIF_BLOCKED_ACCESS_UNTIL_DATE, whitelisted list of certification centers
+# presence: optional
+# type: String (comma-separated list of IDs)
+# PIX_CERTIF_BLOCKED_ACCESS_WHITELIST=123,456,789
```
enlever le # devant

lego-technix reviewed (approved):
✅ Lu et validé avec succès pour l'équipe Accès

AndreiaPena commented on certif/translations/en.json:
```diff
@@ -571,7 +571,8 @@
             "incorrect-data": "The session number and/or the session password entered are incorrect.",
-            "mandatory-fields": "The fields “Session number” and “Session password” are required."
+            "mandatory-fields": "The fields “Session number” and “Session password” are required.",
+            "pix-certif-access": "Site under maintenance, your Pix Certif space will reopen on {date}."
```
todo : avoir un message commun entre l'espace surveillant et le centre de certif

AndreiaPena committed:
🔥 certif: remove unused getter

Steph0 committed:
:sparkles: api: fetch center autorisations through internal API

Agnes-V reviewed (approved):
Yo @AndreiaPena !
Func review nickel 👌✅
Est-ce que tu peux modifier de nouveau la date de fin de période de la variable d'environnement PIX_CERTIF_BLOCKED_ACCESS_UNTIL_DATE=2020-01-01 stp ? 🙏

Steph0 commented on api/src/identity-access-management/infrastructure/repositories/certification-point-of-contact.repository.js:
```diff
@@ -7,13 +7,79 @@ import { CertificationPointOfContact } from '../../domain/read-models/Certificat
+ * @param {number} certificationCenterId - List of certification centers.
+ * @returns {Promise<AllowedCertificationCenterAccess>} - center access
+ */
+const getCertificationCenterAccess = async ({ certificationCenterId }) => {
```
ℹ️ L'autre fonction possède de grosses lacunes et ne correspond pas au besoin (besoin d'un seul centre). Cette version fix les flaws en particulier sur l'aspect toDomain.

yaf commented on api/src/certification/evaluation/domain/models/Session.js:
```diff
@@ -1,9 +1,16 @@
+export class Session {
+  constructor({ id, accessCode, finalizedAt, publishedAt } = {}) {
     this.id = id;
+    this.accessCode = accessCode;
```
question : je ne comprends pas très bien ce que vient faire cette histoire d'`accessCode` dans cette PR ? J'ai du rater un truc :)

Steph0 replied:
Oui, pour expliquer, on utilisait avant un repository (et du coup son domain model) d'un **autre** context (violation de contexte), au lieu du domain model de son propre contexte. Cela causait une dep circulaire.

Du coup on a corrigé cela, et le nouveau repo utilise bien le domain model de son contexte. Il suffisait de remonter l'accesscode pour pouvoir utiliser le model local qui existait deha, hence cet ajout.

Steph0 replied:
pqreil pour le isAccessible d'ailleurs

yaf commented:
J'ai beau avoir initié cette PR, le code ne ressemble plus trop à ce que j'avais fait au début. Aussi, j'ai fait une revue, et ça me semble très bien !

Agnes-V commented:
Yo @yaf ou @Steph0 ou @lgoubin ou @Jeyffrey !
Func review nickel 👌✅
Est-ce que vous pouvez modifier la date de fin de période de la variable d'environnement PIX_CERTIF_BLOCKED_ACCESS_UNTIL_DATE=2020-01-01 svp ? 🙏

lgoubin commented:
> Yo @yaf ou @Steph0 ou @lgoubin ou @Jeyffrey ! Func review nickel 👌✅ Est-ce que vous pouvez modifier la date de fin de période de la variable d'environnement PIX_CERTIF_BLOCKED_ACCESS_UNTIL_DATE=2020-01-01 svp ? 🙏

C'est fait

Agnes-V reviewed (approved):
Func review OK avec changement date de blocage ✅

Steph0 reviewed (approved):
J'ai du corriger les tests suite disparition `hasComplementaryReferential` on envoie cette PR qui n'a que trop attendue

pix-service commented:
:tada: This PR is included in version 5.218.0 :tada:

The release is available on [GitHub release](https://github.com/1024pix/pix/releases/tag/v5.218.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
